packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/useAsyncFunctionCall.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
+
+type AsyncFunc<ReturnType> = (signal: AbortSignal) => Promise<ReturnType>;
+type AsyncFuncReturnType<T> = T extends AsyncFunc<infer U> ? U : never;
+
+/**
+ * Call async function WRAPPED BY `useCallback` and return its result
```

Comment:
So every instance of this _requires_ that the caller use `useCallback`. Why not just bring that into this hook? And call it `useAsyncCallback`.

Reply:
I tried it but I found it needs some hack. So I choose this way to keep the React hook's way.

```typescript
// caller
useAsyncFunc(async (arg1,arg2) => {body;}, [arg1,arg2]);
// hook
function useAsyncCallback(asyncFunc, args) {
  useEffect(() => {
     // run await asyncFunc(...args);
     
     // disable hook dependency warning
  }, [asyncFunc, ...args]);
}
```

Reply:
Gotcha. Thanks for the explanation.